week-3/D-array-filter/exercise.js

```diff
@@ -8,15 +8,13 @@
 
 var pairsByIndexRaw = [[0, 3], [1, 2], [2, 1], null, [1], false, "whoops"];
 
-var pairsByIndex; // Complete this statement
+var pairsByIndex = pairsByIndexRaw.map(function(indexes) {
```

Comment:
This won't work on some of the inputs. For example, null, false, [1], and "whoops". You need to filter those inputs first.
```suggestion
function isPair(item) {
  return Array.isArray(item) && (item.length == 2);
}

var pairsByIndex = pairsByIndexRaw.filter(isPair);
```